java/amazon-kinesis-producer-sample/src/com/amazonaws/services/kinesis/producer/sample/SampleProducer.java

```diff
@@ -213,12 +290,13 @@ public static void main(String[] args) throws Exception {
             @Override
             public void onFailure(Throwable t) {
                 // If we see any failures, we will log them.
+                int attempts = ((UserRecordFailedException) t).getResult().getAttempts().size()-1;
                 if (t instanceof UserRecordFailedException) {
-                    Attempt last = Iterables.getLast(
-                            ((UserRecordFailedException) t).getResult().getAttempts());
+                    Attempt previous = ((UserRecordFailedException) t).getResult().getAttempts().get(attempts-1);
```

Comment:
What happens when getAttempts().size = 1 ?
Doesnt that cause this line to try to get(-1) ?

Reply:
Addressed.